src/components/PostDetails/PostDetails.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react';
+import { NewCommentForm } from '../NewCommentForm';
+import './PostDetails.scss';
+import { Post, Comment } from '../../types';
+import { getComments, delComment } from '../../api/comments';
+import { getPostbyId } from '../../api/posts';
+
+type Props = {
+  postId: number | undefined;
+};
+
+export const PostDetails: React.FC<Props> = ({
+                                               postId,
+                                             }) => {
+  const [commentsList, setCommentsList] = useState<Comment[] | undefined>([]);
+  const [visiblecomments, setVisiblecomments] = useState(false);
+  const [postDetails, setPostDetails] = useState<Post>();
+
+  const findPost = async () => {
+    if (postId) {
+      const result = await getPostbyId(postId);
+
+      setPostDetails(result);
+    }
+  };
+
+  useEffect(() => {
+    findPost();
+  }, []);
+
+  const findcomments = async () => {
+    if (postId) {
+      const result = await getComments(postId);
+
+      setCommentsList(result);
+    }
+  };
+
+  const deletecomment = async (id: number) => {
+    if (commentsList) {
+      await delComment(id);
+      findcomments();
+    }
+  };
+
+  useEffect(() => {
+    findcomments();
+  }, [postId]);
```

Comment:
Here we call a method that is called in more than one place, so we don't need to remove it from the component.

but we still need to wait for its response, so we should wrap it into a async ([IIFE](https://developer.mozilla.org/pt-BR/docs/Glossary/IIFE)) or insert a async function inside this (and any other that makes requests) useEffect.